libs/json/src/document.cpp

```diff
@@ -71,7 +71,7 @@ void JSONDocument::ExtractPrimitive(Variant &variant, JSONToken const &token,
   case NUMBER_INT:
   {
     std::string const str{document.SubArray(token.first, token.second)};
-
+    errno   = 0;
```

Comment:
Is this strictly necessary?

Reply:
Yes it is, and I discovered it on my linux installation while executing unit tests. C/C++ standard explicitly says for all `strtoXX(...)` / `std::strtoXX(...)` functions that the `errno` global variable is set when something goes wrong (errno value remains unchanged when conversion is successful).

Reply:
I would say so. I discovered the issue on one of my dev setups (either Linux of Mac, can remember which one) while executing unit tests. C/C++ standard does not enforce setting `errno` global variable for any of `strtoXX(...)` / `std::strtoXX(...)` functions if conversion is successful, it only defines that the the `errno` global variable is set when something goes wrong.
[Here](https://stackoverflow.com/a/14176593) is a little bit of reading on the topic.